lib/rustici_software_cloud_v2/api_client.rb

```diff
@@ -264,7 +264,7 @@ def sanitize_filename(filename)
     def build_request_url(path)
       # Add leading and trailing slashes to path
       path = "/#{path}".gsub(/\/+/, '/')
-      URI.encode(@config.base_url + path)
+      URI.parse(@config.base_url + path).to_s
```

Comment:
I don't think your change is going to provide the escaping that the previous method did.
URI::encode was just an alias for URI::escape.
And assuming Ruby v3.0.0 did change the behavior of URI::parse and as well as URI::to_s to add escaping, this change would not be backward compatible with Ruby's < v3.

Reply:
There doesn't appear to be a 1-to-1 replacement for `URI.encode` in Ruby anymore, but I've found a dependency called [Addressable](https://github.com/sporkmonger/addressable) that seems to have taken up this mantle though if you would be interested in adding it.

As for the code above, `URI.parse` has been in Ruby for a long time, so I don't believe there would be a backward-compatibility issue with it.

Reply:
Just pushed a commit that includes the Addressable dependency so I could test it. It seems to be working fine with my setup

Reply:
I don't actually have any issue with URI::parse.  It just breaks up the URI into components.  URI.to_s just puts them back together.  My point is doing URI.parse(url).to_s doesn't actually produce anything different than the original in terms of escaping any non-URI compatible characters.
This is a good article on this issue: https://medium.com/@rainerborene/7-ways-to-encode-urls-in-ruby-c15078a7d6bc

Reply:
I came across that same article. My assumption was that parsing it with `URI.parse` would ensure the URL is properly formatted in the even there were any formatting problems. I'm not sure that you could just leave it as `URI.parse` because it would the return a URI object, as opposed to a string, which is what I'm assuming is the expected return type.

Another approach could be to `CGI.parse` the path portion of the URL and then concat that string, but I'm not certain that would give you your desired outcome. We can't `CGI.parse` the entire URL as it will encode parts of the URL you don't want, like the colon.

As I mentioned above, the current version this adds the `Addressable` dependency, which seems to be some what of the de facto replacement for `URI.encode`, but if you prefer to take another approach just let me know.

Reply:
And interestingly, this reference here: https://rubyapi.org/3.0/o/uri#method-c-parse
Still mentions: "It's recommended to first ::escape the provided uri_str if there are any invalid URI characters."
Hmm, I guess that perhaps is a left-over?

Reply:
I'm definitely not the one to approve any of this.  I'm just not going to v3 anytime soon, so just looking to keep my stuff working and providing some feedback on maintaining backward compatibility.   I would agree that using Addressable is probably the best approach.  I do wish that Ruby would just incorporate Addressable into the stdlib if that's what they want for the preferred approach going forward.

Reply:
I agree. It’s very frustrating that they would pull a feature without some kind of proposed replacement.